pkg/juicefs/juicefs.go

```diff
@@ -259,81 +287,61 @@ func (j *juicefs) AuthFs(secrets map[string]string, extraEnvs map[string]string)
 		"accesskey2",
 		"bucket",
 		"bucket2",
+		"subdir",
 	}
 	keysStripped := []string{
 		"token",
 		"secretkey",
 		"secretkey2",
 		"passphrase"}
 	isOptional := map[string]bool{
+		"accesskey":  true,
 		"accesskey2": true,
+		"secretkey":  true,
 		"secretkey2": true,
 		"bucket":     true,
 		"bucket2":    true,
 		"passphrase": true,
+		"subdir":     true,
 	}
 	for _, k := range keys {
 		if !isOptional[k] || secrets[k] != "" {
 			args = append(args, fmt.Sprintf("--%s=%s", k, secrets[k]))
-			argsStripped = append(argsStripped, fmt.Sprintf("--%s=%s", k, secrets[k]))
 		}
 	}
 	for _, k := range keysStripped {
 		if !isOptional[k] || secrets[k] != "" {
-			args = append(args, fmt.Sprintf("--%s=%s", k, secrets[k]))
-			argsStripped = append(argsStripped, fmt.Sprintf("--%s=[secret]", k))
+			args = append(args, fmt.Sprintf("--%s=${%s}", k, k))
 		}
 	}
 	if v, ok := os.LookupEnv("JFS_NO_UPDATE_CONFIG"); ok && v == "enabled" {
 		args = append(args, "--no-update")
-		argsStripped = append(argsStripped, "--no-update")
-
 		if secrets["bucket"] == "" {
-			return nil, status.Errorf(codes.InvalidArgument,
+			return "", status.Errorf(codes.InvalidArgument,
 				"bucket argument is required when --no-update option is provided")
 		}
-		if secrets["initconfig"] != "" {
-			conf := secrets["name"] + ".conf"
-			confPath := filepath.Join("/root/.juicefs", conf)
-			if _, err := os.Stat(confPath); os.IsNotExist(err) {
-				err = ioutil.WriteFile(confPath, []byte(secrets["initconfig"]), 0644)
-				if err != nil {
-					return nil, status.Errorf(codes.Internal,
-						"Create config file %q failed: %v", confPath, err)
-				}
-				klog.V(5).Infof("Create config file: %q success", confPath)
-			}
-		}
-	}
-	klog.V(5).Infof("AuthFs: cmd %q, args %#v", config.CliPath, argsStripped)
-	authCmd := j.Exec.Command(config.CliPath, args...)
-	envs := syscall.Environ()
-	for key, val := range extraEnvs {
-		envs = append(envs, fmt.Sprintf("%s=%s", key, val))
 	}
-	klog.V(5).Infof("AuthFs: envs %q", extraEnvs)
-	authCmd.SetEnv(envs)
-	return authCmd.CombinedOutput()
+
+	cmd := strings.Join(args, " ")
+	klog.V(5).Infof("AuthFs: cmd %s", cmd)
```

Comment:
There are secrets in cmd

Reply:
Secret in this cmd is replaced by string `$(xxx)`, this log will not display secret